### PR TITLE
ADD HTTP HEADERS TO EXTRACTIONS: As Dan, I want to be able to add http headers to an extraction definition, so that I can work with NLNZcat and others

### PR DESCRIPTION
**Pull request by richardmatthewsdev**
**Acceptance Criteria**
- Extractions can be configured to use things like:
 - `http_headers({'X-Forwarded-For': 'ab.cd.ef.gh'})` 
 - `http_headers({'Authorization': 'Bearer BLAAAAAAH'})` 
 - `http_headers({'Authorization': 'Basic BLAAAAAAAH'})` 
 - `http_headers({'x-api-key': 'asdsads'})`
 - 
- Allow multiple headers to be added

**Risks**
- ?

### CODE REVIEW

File: spec/models/header_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    subject { build(:header) }
+
+    it { is_expected.to validate_presence_of(:name) }
+    it { is_expected.to validate_presence_of(:value) }
+  end
+
+  describe 'associations' do
+    it 'belongs to an extraction_definition' do
+      expect(subject.extraction_definition).to eq extraction_definition
+    end
```
Comment (hapiben) [inline spec/models/header_spec.rb:17]:
```suggestion
    it { is_expected.belong_to(:extraction_definition) }
```

**Review by hapiben — APPROVED** [review]

File: spec/models/header_spec.rb (hunk 1 above)
Comment (richardmatthewsdev) [reply spec/models/header_spec.rb:17]: Thanks @hapiben !

File: app/frontend/js/ExtractionDefinitionFormLogic.js [hunk 2]
```diff
@@ -33,6 +33,52 @@ if(extractionDefinitionPaginationTypeSelect) {
...
+          <div class='col-4'>
+            <label class='form-label' for='extraction_definition_headers_attributes_${newId}_value'>
+              Header Value
+            </label>                
+          </div>
+        
+          <div class='col-8'>
+            <input class='form-control' type='text' name='extraction_definition[headers_attributes][${newId}][value]'' id='extraction_definition_headers_attributes_${newId}_value'>
+          </div>
+        </div>
```
Comment (paul-mesnilgrente) [inline app/frontend/js/ExtractionDefinitionFormLogic.js:74]: Is it time to switch this form to react?

**Review by paul-mesnilgrente — APPROVED** [review]

File: app/frontend/js/ExtractionDefinitionFormLogic.js (hunk 2 above)
Comment (richardmatthewsdev) [reply app/frontend/js/ExtractionDefinitionFormLogic.js:74]: Yes I think so. When we do the `PAGINATION ADJUSTMENTS` story I think that will be a good opportunity to replace this form with React.

**Pull request merged**